internal/dag/ingress_processor.go

```diff
@@ -165,7 +175,7 @@ func (p *IngressProcessor) computeIngressRule(ing *v1beta1.Ingress, rule v1beta1
 }
 
 // route builds a dag.Route for the supplied Ingress.
-func route(ingress *v1beta1.Ingress, path string, service *Service, clientCertSecret *Secret, log logrus.FieldLogger) (*Route, error) {
+func route(ingress *networking_v1.Ingress, path string, service *Service, clientCertSecret *Secret, log logrus.FieldLogger) (*Route, error) {
```

Comment:
Wondering what we should do with the [pathType](https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types) field for now. It looks like the main issue would be if a user has explicitly specified `pathType: Exact`, Contour would currently still treat it as a prefix. We have https://github.com/projectcontour/contour/issues/2135 to implement that properly; perhaps until then we should return an error if a `pathType: Exact` is encountered?

Reply:
per discussion, Contour is already ignoring this field and treating everything as either a regex or a prefix, so it's OK to continue doing that until we implement #2135.